### PR TITLE
Cleanup federation phase 1

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
The first part of federatedfileshare code cleanup
- Get rid of some deprecated classes in the code
- Get rid of some unused classes in the code
- Cleanup routing
- Provides an ability to bypass changing `/ocs/routes.php` and set up absolute ocs routes in the `appinfo/routes.php` 

## Related Issue
https://github.com/owncloud/enterprise/issues/2564

## Motivation and Context
Clean and readable code 

## How Has This Been Tested?
1. Created an incoming share from another instance at master branch
2. Accepting|Declining|Unsharing it

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

## Checklist:
- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: apps/federatedfilesharing/lib/FedShareManager.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+		try {
+			$file = \OC\Files\Filesystem::getPath($fileSource);
+		} catch (NotFoundException $e) {
+			$file = null;
+		}
+		$args = \OC\Files\Filesystem::is_dir($file)
+			? ['dir' => $file]
+			: ['dir' => \dirname($file), 'scrollto' => $file];
+		$link = \OCP\Util::linkToAbsolute('files', 'index.php', $args);
```
Comment (PVince81) [inline apps/federatedfilesharing/lib/FedShareManager.php]: note for later: we could use permalink here, see ViewController how to generate one

File: apps/federatedfilesharing/lib/RequestHandler.php [hunk 2]
```diff
@@ -286,19 +287,17 @@ public function acceptShare($params) {
...
-				$this->notifications->sendAcceptShare($remote, $remoteId, $share->getToken());
+			if ($this->verifyShare($share, $token)) {
+				$this->fedShareManager->acceptShare($share);
+				if ($share->getShareOwner() !== $share->getSharedBy()) {
+					list(, $remote) = $this->addressHandler->splitUserRemote($share->getSharedBy());
+					$remoteId = $this->federatedShareProvider->getRemoteId($share);
+					$this->notifications->sendAcceptShare($remote, $remoteId, $share->getToken());
+				}
 			}
+		} catch (\Exception $e) {
```
Comment (PVince81) [inline apps/federatedfilesharing/lib/RequestHandler.php]: better catch Share\Exceptions\ShareNotFound then and let the other through for proper error reporting ?

File: apps/federatedfilesharing/lib/RequestHandler.php [hunk 3]
```diff
@@ -318,17 +317,16 @@ public function declineShare($params) {
...
+			if ($this->verifyShare($share, $token)) {
+				if ($share->getShareOwner() !== $share->getSharedBy()) {
+					list(, $remote) = $this->addressHandler->splitUserRemote($share->getSharedBy());
+					$remoteId = $this->federatedShareProvider->getRemoteId($share);
+					$this->notifications->sendDeclineShare($remote, $remoteId, $share->getToken());
+				}
+				$this->fedShareManager->declineShare($share);
 			}
-			$this->fedShareManager->declineShare($share);
+		} catch (\Exception $e) {
```
Comment (PVince81) [inline apps/federatedfilesharing/lib/RequestHandler.php]: same here

File: apps/files_sharing/ajax/external.php [hunk 4]
```diff
@@ -32,7 +32,7 @@
 
 $l = \OC::$server->getL10N('files_sharing');
 
-$federatedSharingApp = new \OCA\FederatedFileSharing\AppInfo\Application('federatedfilesharing');
+$federatedSharingApp = new \OCA\FederatedFileSharing\AppInfo\Application();
```
Comment (PVince81) [inline apps/files_sharing/ajax/external.php]: sad to see we still have ajax endpoints, oh well

**Review by PVince81 — CHANGES_REQUESTED** [review]
Awesome!

Please adjust the exception catch block I pointed to

**Comment by VicDeo** [comment]
@PVince81 updated

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
As discussed, unit tests will be added in the next round

**Comment by PVince81** [comment]
no backport for now

**Comment by VicDeo** [comment]
Stable10: https://github.com/owncloud/core/pull/33027

**Pull request merged**